source/pull-request.rst

```diff
@@ -0,0 +1,237 @@
+
+.. dropdown:: :fa:`exclamation-circle,mr-1` 安装 Python、Sphinx 以及文档所需依赖包
+   :container: + shadow
+   :title: bg-info text-white font-weight-bold 
+
+    地震“学”的所有文档均使用文档生成工具 `Sphinx <http://www.sphinx-doc.org/>`__
+    构建。因此，若想要在本地构建文档并检查修改效果，需要安装 Python、Sphinx 以及文档
+    所需依赖包：
+
+    - 建议通过 :doc:`Anaconda <software:anaconda/index>` 来安装和管理 Python。
+      强烈建议不要使用系统自带的 Python。
+    - 参考\ :doc:`building`\ ，安装 Sphinx 和文档所需依赖包。
+
+克隆和复制仓库
+^^^^^^^^^^^^^^
+
+.. note::
+
+    以下步骤假定读者的 GitHub 用户名为 ``seismology-freshman``\ 。
+
+    以下步骤以\ `地震“学”科研入门教程 <https://seismo-learn.org/seismology101/>`__\ 为例，修改其他文档时做简单替换即可：
+
+    - 修改文档\ `地震“学”软件 <https://seismo-learn.org/software/>`__\ : 将 ``seismology101`` 改成 ``software``
+    - 修改文档\ `地震“学”参考书 <https://seismo-learn.org/seismology/>`__\ : 将 ``seismology101`` 改成 ``seismology``
+
+1.  复制仓库至个人 GitHub 帐号
+
+    点击项目主页 https://github.com/seismo-learn/seismology101 右上角的 Fork 按钮，将该项目复制到个人 GitHub 账户下。
+    复制完成后，个人 GitHub 帐号下便有了 https://github.com/seismology-freshman/seismology101 仓库。
+
+2.  克隆个人 GitHub 帐号下的复制仓库到本地计算机（复制仓库默认是本地克隆仓库的远程 origin）::
+
+        # 进入 ~/Downloads 目录，也可以选择其他目录存放文档源码
+        $ cd ~/Downloads
+
+        # 克隆仓库
+        $ git clone git@github.com:seismology-freshman/seismology101.git
+
+3.  添加官方仓库作为本地克隆仓库的另一个远程，并命名为 upstream::
+
+        # 进入 ~/Downloads/seismology101 目录
+        $ cd ~/Downloads/seismology101/
+
+        # 添加官方仓库作为另一个远程 upstream
+        $ git remote add upstream git@github.com:seismo-learn/seismology101.git
+
+.. note::
+
+   上述三个步骤只需在第一次修改代码时执行一次。一旦复制或克隆某仓库后，就无需再次复制或克隆。
+
+修改文档
+^^^^^^^^
+
+1.  在本地创建并切换至新分支，假定新分支名为 ``pr-workflow``\ （分支名需简短、描述性且独特）::
+
+        # 若不在 main 分支，需先切换至该分支
+        $ git checkout main
+
+        # 创建并切换至 pr-workflow 分支
+        $ git checkout -b pr-workflow
+
+2.  在新建分支中对文档做修改，并提交 commit（此过程可以循环多次）::
+
+        # 查看仓库当前的状态
+        $ git status
+        # 添加所有修改
+        $ git add --all
+
+        # 查看仓库当前的状态
+        $ git status
+        # 提交添加的修改
+        $ git commit -m "此处填写本次提交的注释信息"
+
+    .. warning::
+
+       切忌不要直接在 main 分支中进行修改和提交
+
+3.  修改过程中，可以随时在本地构建文档并检查修改效果
+
+    使用以下命令构建文档，生成的文档位于本地仓库的 :file:`build/html/` 目录下，直接用浏览器打开
+    :file:`build/html/index.html` 即可预览::
+
+        # 进入仓库主目录
+        $ cd ~/Downloads/seismology101/
+
+        # 编译生成 HTML 格式的文档
+        $ make html
+
+.. note::
+
+   开发 pr-workflow 分支的过程中，官方 main 分支可能已经更新。因此，需要经常同步最新版的官方 main 分支。
+
+   1.  同步本地和官方 main 分支::
+
+           # 切换到 main 分支
+           $ git checkout main
+
+           # 获取官方 main 分支，并合并到本地 main 分支
+           $ git pull upstream main
+
+   2.  更新个人 GitHub 帐号下的复制仓库 main 分支::
+
+           $ git push origin main
+
+   3.  将 pr-workflow 分支基于最新的 main 分支::
+
+           # 切换到 pr-workflow 分支
+           $ git checkout pr-workflow
+
+           # 将 pr-workflow 分支基于最新的 main 分支
+           $ git rebase main
+
+.. note::
+
+   分支开发的过程中，可能会有很多次 commit，某些 commit 可能不那么重要。可以将多个 commit
+   压缩成一个或若干个 commit，这样不仅清晰，也容易管理::
+
+       $ git rebase -i main
```

Comment:
Perhaps remove this note. When we merge a PR, we would squash it, so the messy commits don't matter.

Reply:
Removed in 0872e6a.